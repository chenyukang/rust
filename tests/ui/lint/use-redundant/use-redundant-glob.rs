//@ check-pass
#![warn(unused_imports)]

pub mod bar {
    pub struct Foo(pub Bar);
    pub struct Bar(pub char);
}

pub fn warning() -> bar::Foo {
    use bar::Foo; //~ WARNING redundant import: `bar::Foo`
    use bar::*;
    Foo(Bar('a'))
}

fn main() {}
