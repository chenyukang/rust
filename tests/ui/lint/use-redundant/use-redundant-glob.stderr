warning: redundant import: `bar::Foo`
  --> $DIR/use-redundant-glob.rs:10:9
   |
LL |     use bar::Foo;
   |         ^^^^^^^^
LL |     use bar::*;
   |         ------ the item `Foo` is already imported here
   |
note: the lint level is defined here
  --> $DIR/use-redundant-glob.rs:2:9
   |
LL | #![warn(unused_imports)]
   |         ^^^^^^^^^^^^^^

warning: 1 warning emitted

